### ISSUE TITLE
Avoid accidental @-mentions in transcripts

### ISSUE THREAD

**Issue by AmeliaBR** [dbaron/wgmeeting-github-ircbot#52]
See https://github.com/w3c/csswg-drafts/issues/4968#issuecomment-617888183 for an example of problems that this causes. (I've since edited that comment to break the mentions.)

Someone in the chat used an @ and the IRC nick of a group member, but that nick belongs to an unrelated person who was annoyed by the sudden notification spam!

In the same chat, there were multiple instances of @ being used in the context of CSS rules (@ media and @ supports), and they were also getting automatically upgraded into GitHub mentions, although those seem to be unused accounts.

So, my suggestion would be to run a find & replace to insert a space or something after @ characters in the transcript.

**Comment by upsuper** [comment]
Alternatively, we can wrap any @-mention with inline code (backtick) so that it's not upgraded, and we don't break the word with space.

**Comment by nigelmegitt** [comment]
I use `@` notation frequently as a deliberate way to mention people, and make sure to use their GitHub name not their IRC name, so disabling that would be a pain. I agree that it's no fun to have someone's IRC nick used as their GitHub name when it's someone different though!

I'd like something that's likely much more complex to implement, which is that `@` notation followed by GitHub name is retained as now, but that some other notation is used to bring IRC nick's in and (this is the hard part) map it to their pre-registered GitHub name.

So say I'm `nigel` on IRC, and @nigelmegitt on GitHub, when scribing I might do something like:

```
.. and everyone really loved the excellent proposal from $nigel
```

and github-bot would replace `$nigel` with `@nigelmegitt` before posting the issue comment. This would also be super-handy when you can't remember someone's GitHub name, and mean that the scribe could use IRC-nick-based autocomplete, and the IRC log would match the list of those present.

I'm not sure if IRC nicks are registered anywhere though; I think probably not, so that would imply either some additional backend service is needed or at the start of each meeting an additional command would be needed per user, like:

```
github: nigel is @nigelmegitt 
```

and GitHub-bot would have to remember that for some period of time.

**Comment by dbaron** [comment]
Two comments here:
* One of the tradeoffs here is that I've tried to make the bot's github comments be at least somewhat readable in their plain-text email form, although I admit it's not great.  Some of the suggestions here might make things worse.
* Second, the IRC log within the bot comments is within a markdown [HTML block](https://github.github.com/gfm/#html-blocks).  Some of the suggestions here (backticks) don't work in markdown's HTML blocks.  (It's entirely possible to have the contents of a `<details>` *not* be in markdown's HTML blocks -- it just requires blank lines -- and I did experiment with that in the early days of the bot -- but it would require redesigning a bunch of what the bot does today.  In particular, the current design makes the escaping/sanitizing requirements for the bot substantially simpler than they would be if producing markdown outside of an HTML block.)

**Comment by dbaron** [comment]
Oh, and I think that replacing "@" with "@ " would be problematic because it would break people's syntax examples; it would be confusing to read a discussion about CSS at-rule syntax with such a substitution.

**Comment by dbaron** [comment]
Yeah, that's probably the least bad option (so far) that would fix this, though it shows up in plain text email, and some folks (see https://github.com/dbaron/wgmeeting-github-ircbot/issues/52#issuecomment-618277284) use this as a feature rather than a bug.
